### PR TITLE
Submission week04

### PR DESCRIPTION
**Pull request by fifo36**

### CODE REVIEW

File: week04/1268. Search Suggestions System.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        curNode->nodeWord = word;
+    }
+
+    void getSuggestedWords(trieNode *curNode, vector<string> &vec){
+        if(curNode == NULL) return;
+        if(curNode->nodeWord.size()) vec.push_back(curNode->nodeWord);
+
+        for(int i = 0; i < 26; i++){
+            if(vec.size() < 3) getSuggestedWords(curNode->next[i], vec);
+        }
```
Comment (Sadman007) [inline week04/1268. Search Suggestions System.cpp:33]:
```suggestion
        for(int i = 0; i < 26; i++){
            if(curNode->next[i] == NULL) continue;
            if(vec.size() < 3) getSuggestedWords(curNode->next[i], vec);
        }
```

Comment (Sadman007) [reply week04/1268. Search Suggestions System.cpp:33]: This should improve the runtime. (Think why?)

File: week04/139. Word Break.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if(pos >= s.size()) return true;
+        if(vis[pos]) return dp[pos];
+        bool curResult = 0;
+        for(int i = pos, len = 1; s[i] && len <= 20; i++, len++){
+            curResult |= (checkTrie(pos, len, s, rootNode) & getResult(pos+len, s, rootNode));
+        }
+        vis[pos] = 1;
+        return dp[pos] = curResult;
+    }
+
```
Comment (Sadman007) [inline week04/139. Word Break.cpp:53]: Nice but not the optimal one.

**Review by Sadman007 — APPROVED** [review]
Hello @fifo36 

Nice work!

I have completed my review. Feel free to close/merge this PR. Thanks.

File: week04/139. Word Break.cpp (hunk 2 above)
Comment (fifo36) [reply week04/139. Word Break.cpp:53]: Etato 0sec, 5sec, 9sec erokom time e pass hoy. Tobuo optimal na?

Comment (Sadman007) [reply week04/139. Word Break.cpp:53]: Jokhon review korsilam tokhon ki code etai chilo? I doubt. Amar mone hoy different ekta dekhsilam. Eta optimal jeta ekhon dekhtesi.

Anyway, etar mark aagei paiso. Kata hoynai

**Comment by fifo36** [comment]
done

**Pull request closed without merging**